presto-client/src/main/java/com/facebook/presto/client/StatementClientV1.java

```diff
@@ -309,6 +309,17 @@ public boolean advance()
         long attempts = 0;
 
         while (true) {
+            Duration sinceStart = Duration.nanosSince(start);
+            if (sinceStart.compareTo(requestTimeoutNanos) > 0) {
+                state.compareAndSet(State.RUNNING, State.CLIENT_ERROR);
+                // cause may be null here if requestTimeoutNanos is extremely small.
```

Comment:
Let's put this inside the `if (attempts > 0)` block so we never fail with `"attempts: 0"` which would be lame. Also, technically, there could be a JVM / process schedule pause that causes us to hit the timeout here without making any attempt.